botocore/handlers.py

```diff
@@ -346,6 +346,14 @@ def add_expect_header(model, params, **kwargs):
     if 'body' in params:
         body = params['body']
         if hasattr(body, 'read'):
+            check_body = utils.ensure_boolean(
+                os.environ.get(
+                    'BOTO_EXPERIMENTAL__CHECK_BODY_BEFORE_CONTINUE',
```

Comment:
nit; the name could be a bit more succinct.
```suggestion
                    'BOTO_EXPERIMENTAL__NO_EMPTY_CONTINUE',
```

or

```suggestion
                    'BOTO_EXPERIMENTAL__NO_EMPTY_100',
```